pkg/io/video/convert_cgo_test.go

```diff
@@ -0,0 +1,84 @@
+// +build cgo
+
+package video
+
+import (
+	"image/color"
+	"testing"
+)
+
+func diffUint8(a, b uint8) uint8 {
+	d := int(a) - int(b)
+	if d < 0 {
+		d = -d
+	}
+	return uint8(d)
+}
+
+func TestRGBToYCbCr(t *testing.T) {
+	for r := 0; r < 0x100; r += 8 {
+		for g := 0; g < 0x100; g += 8 {
+			for b := 0; b < 0x100; b += 8 {
+				var y, cb, cr uint8
+				rgbToYCbCrCGO(&y, &cb, &cr, uint8(r), uint8(g), uint8(b))
+				y2, cb2, cr2 := color.RGBToYCbCr(uint8(r), uint8(g), uint8(b))
+				if diffUint8(y, y2) > 2 || diffUint8(cb, cb2) > 2 || diffUint8(cr, cr2) > 2 {
+					t.Fatalf(
+						"rgbToYCbCrCGO differs from color.RGBToYCbCr: in(%d, %d, %d), "+
+							"expected(%d, %d, %d), got(%d, %d, %d)",
+						r, g, b,
+						y2, cb2, cr2,
+						y, cb, cr,
+					)
+				}
+			}
+		}
+	}
+}
+
+func TestYCbCrToRGB(t *testing.T) {
+	for y := 0; y < 0x100; y += 8 {
+		for cb := 0; cb < 0x100; cb += 8 {
+			for cr := 0; cr < 0x100; cr += 8 {
+				var r, g, b uint8
+				yCbCrToRGBCGO(&r, &g, &b, uint8(y), uint8(cb), uint8(cr))
+				r2, g2, b2 := color.YCbCrToRGB(uint8(y), uint8(cb), uint8(cr))
+				if diffUint8(r, r2) > 2 || diffUint8(g, g2) > 2 || diffUint8(b, b2) > 2 {
+					t.Fatalf(
+						"yCbCrToRGBCGO differs from color.YCbCrToRGB: in(%d, %d, %d), "+
+							"expected(%d, %d, %d), got(%d, %d, %d)",
+						y, cb, cr,
+						r2, g2, b2,
+						r, g, b,
+					)
+				}
+			}
+		}
+	}
+}
+
+func BenchmarkRGBToYCbCr(b *testing.B) {
+	b.Run("Go", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			_, _, _ = color.RGBToYCbCr(200, 100, 10)
+		}
+	})
+	b.Run("CGO", func(b *testing.B) {
+		var y, cb, cr uint8
+		// Loop in C code To ignore CGO call overhead
+		repeatRGBToYCbCrCGO(b.N, &y, &cb, &cr, 200, 100, 10)
```

Comment:
Hmm, I don't think this is a fair benchmark with the Go version since we always need to have the loop in Go due to the interface that we currently have. What do you think?

Reply:
CGO version of RGBToYCbCr is basically used only from [C loop](https://github.com/pion/mediadevices/pull/67/files#diff-fd946c9906a76d5c7820d3c00f70b38dR168), and used in Go only for testing.
I think it's not possible to compare such small C code and Go code due to the overhead. I just wanted to check it is not very slow compared with stdlib image package.

Reply:
In my experience of these a few days, this kind of calculation is not a bottleneck.
The main bottleneck seems be iterations scanning Go slice.

Reply:
> CGO version of RGBToYCbCr is basically used only from [C loop](https://github.com/pion/mediadevices/pull/67/files#diff-fd946c9906a76d5c7820d3c00f70b38dR168), and used in Go only for testing.
> I think it's not possible to compare such small C code and Go code due to the overhead. I just wanted to check it is not very slow compared with stdlib image package.

Oops, I'm not sure why. But, when I read the repeat code, I thought that RGBToYCbCrCGO was for converting the whole image... But it's definitely not. It's just per color.

Reply:
> In my experience of these a few days, this kind of calculation is not a bottleneck.
> The main bottleneck seems be iterations scanning Go slice.

That's interesting. It sounds to me that it's more about cache locality. But, you definitely know better than I do for this part...